### PR TITLE
Spelling stdlib

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift documentation, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101456910

### CODE REVIEW

File: stdlib/include/llvm/Support/type_traits.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//===- llvm/Support/type_traits.h - Simplfied type traits -------*- C++ -*-===//
+//===- llvm/Support/type_traits.h - Simplified type traits -------*- C++ -*-===//
```
Comment (jsoref) [inline stdlib/include/llvm/Support/type_traits.h:1]: A bunch of these items exceed 80 chars, but while I could fix this...

File: stdlib/toolchain/Compatibility50/CompatibilityOverride.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//===--- CompatibiltyOverride.h - Back-deploying compatibility fixes --*- C++ -*-===//
+//===--- CompatibilityOverride.h - Back-deploying compatibility fixes --*- C++ -*-===//
```
Comment (jsoref) [inline stdlib/toolchain/Compatibility50/CompatibilityOverride.h:1]: I don't think I could fix this

Comment (CodaFi) [reply stdlib/toolchain/Compatibility50/CompatibilityOverride.h:1]: These headers are kind of best-effort. As long as clang-format doesn't complain it's not a big deal.

**Review by CodaFi — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Pull request merged**